add unit type explanation (#648)
cairo-book/cairo-book: src/ch02-02-data-types.md

It is represented by a tuple with no elements.
Its size is always zero, and it is guaranteed to not exist in the compiled code.

> You might be wondering why you would even need a unit type? In Cairo, functions must always return some value. Even if your expression is not meant to return anything, this will be its default return type. You don't have to explicitly specify it when declaring your functions, but it's good to be clear about why it is even there. You will learn more about functions later in the book.

Reviewer: ```suggestion
You might be wondering why you would even need a unit type? In Cairo, everything is an expression, and an expression that returns nothing actually returns `()`  implicitly.
```
